packages/website/pages/account/payment.js

```diff
@@ -72,51 +47,115 @@ const PaymentSettingsPage = props => {
     const getSavedCard = async () => {
       const card = await getSavedPaymentMethod();
       if (card) {
-        setSavedPaymentMethod(card.method);
+        setSavedPaymentMethod(card.paymentMethod);
       }
-      console.log(card);
       return card;
     };
     getSavedCard();
   }, [hasPaymentMethods]);
 
+  useEffect(() => {
+    if (!currentPlan || currentPlan.id === null) {
+      setPlanList(plansEarly);
+    } else {
+      setPlanList(plans);
+    }
+  }, [currentPlan]);
+
+  useEffect(() => {
+    if (savedPaymentMethod) {
+      const getPlan = async () => {
+        const userPlan = await getUserPaymentPlan();
+        if (userPlan?.subscription?.storage) {
+          try {
+            await setCurrentPlan(planList.find(plan => plan.id === userPlan.subscription.storage.price));
+          } catch {
+            throw new Error('MISSING PLAN');
```

Comment:
@cmunns Is this error more useful than the error it's catching, or was it just there from debugging? If it's from debugging, it seems like this error may have less info than the caught one, so it might be best to throw the original, or otherwise include more useful logging or error message